build fails on osx arm64 (M1) due to use of x86intrinsics header

Attempting to build pycbc on apple arm64 fails as follows:

```
arm64-apple-darwin20.0.0-clang -fno-strict-aliasing -Wsign-compare -Wunreachable-code -DNDEBUG -fwrapv -O3 -Wall -Wstrict-prototypes -ftree-vectorize -fPIC -fPIE -fstack-protector-strong -O3 -pipe -fdebug-prefix-map=${SRC_DIR}=/usr/local/src/conda/${PKG_NAME}-${PKG_VERSION} -fdebug-prefix-map=/Users/runner/miniforge3/conda-bld/pycbc_1633027213478/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla=/usr/local/src/conda-prefix -flto -Wl,-export_dynamic -ftree-vectorize -fPIC -fPIE -fstack-protector-strong -O3 -ftree-vectorize -fPIC -fPIE -fstack-protector-strong -O2 -pipe -isystem /Users/runner/miniforge3/conda-bld/pycbc_1633027213478/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla/include -fdebug-prefix-map=/Users/runner/miniforge3/conda-bld/pycbc_1633027213478/work=/usr/local/src/conda/pycbc-1.18.3 -fdebug-prefix-map=/Users/runner/miniforge3/conda-bld/pycbc_1633027213478/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla=/usr/local/src/conda-prefix -fopenmp -D_FORTIFY_SOURCE=2 -isystem /Users/runner/miniforge3/conda-bld/pycbc_1633027213478/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla/include -mmacosx-version-min=11.0 -I/Users/runner/miniforge3/conda-bld/pycbc_1633027213478/_build_env/venv/lib/python3.8/site-packages/numpy/core/include -I/Users/runner/miniforge3/conda-bld/pycbc_1633027213478/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla/include/python3.8 -c pycbc/filter/simd_correlate_cython.cpp -o build/temp.macosx-11.0-arm64-3.8/pycbc/filter/simd_correlate_cython.o -O3 -w -ffast-math -ffinite-math-only -stdlib=libc++
  In file included from pycbc/filter/simd_correlate_cython.cpp:640:
  In file included from pycbc/filter/simd_correlate_ccode.cpp:36:
  In file included from /Users/runner/miniforge3/conda-bld/pycbc_1633027213478/_build_env/lib/clang/11.1.0/include/x86intrin.h:13:
  /Users/runner/miniforge3/conda-bld/pycbc_1633027213478/_build_env/lib/clang/11.1.0/include/ia32intrin.h:188:10: error: use of undeclared identifier '__builtin_ia32_readeflags_u32'
    return __builtin_ia32_readeflags_u32();
           ^
  /Users/runner/miniforge3/conda-bld/pycbc_1633027213478/_build_env/lib/clang/11.1.0/include/ia32intrin.h:194:3: error: use of undeclared identifier '__builtin_ia32_writeeflags_u32'
    __builtin_ia32_writeeflags_u32(__f);
    
<snip>
```

I presume that x86intrin.h isn't an appropriate header for use on this architecture.

I have attempted to patch this in the conda build [here](https://github.com/conda-forge/pycbc-feedstock/pull/55).

--- a/pycbc/filter/simd_correlate_ccode.cpp
+++ b/pycbc/filter/simd_correlate_ccode.cpp
@@ -35,7 +35,10 @@
 #ifndef __clang__
     #include <omp.h>
 #endif
-#include <x86intrin.h>
+
+#if (defined(__x86_64__) || defined(__i386__))
+    #include <x86intrin.h>
+#endif
 
 #ifdef __AVX2__
 #define _HAVE_AVX2 1

